packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -344,7 +343,9 @@ export class Outbox {
 		// If so, do nothing, as pending state manager will resubmit it correctly on reconnect.
 		// Because flush() is a task that executes async (on clean stack), we can get here in disconnected state.
 		if (this.params.shouldSend()) {
-			const processedBatch = this.compressBatch(rawBatch, disableGroupedBatching);
+			const processedBatch = this.compressBatch(
+				shouldGroup ? this.params.groupingManager.groupBatch(rawBatch) : rawBatch,
```

Comment:
With this change, should the op grouping manager assert instead of returning the batch as-is when it's called to group when it wasn't supposed to?